### PR TITLE
ndctl.py: Fixes and enhancements

### PR DESCRIPTION
**Pull request by harish-24**
1) Add daxctl package for running DAX test
Patch adds daxctl package for running DAX test

2) Fix: Install packages before configuring ndctl
Patch fixes packages installation before configuring ndctl repo

3) Add support to run test on RHEL distribution
Add support to run ndctl test on RHEL distribution

4) Fix: namespace creation size with multiple regions
Current implementation overwrites size on first iteration and uses the same for all regions. hence replacing the variable with a proper persistence.

Signed-off-by: Harish <harish@linux.ibm.com>

### CODE REVIEW

**Comment by harish-24** [comment]
```
# avocado run tests/avocado-misc-tests/memory/ndctl.py
...
 (01/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_bus_ids: PASS (23.27 s)
 (02/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_dimms: PASS (0.75 s)
 (03/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_regions: PASS (0.79 s)
 (04/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_namespace: PASS (34.84 s)
 (05/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_disable_enable_ns: FAIL: Namespace enable failed for "all" (2.65 s)
 (06/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_namespace_modes: FAIL: Namespace for ['raw'] mode failed! (67.51 s)
 (07/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_multiple_namespaces_region: PASS (23.89 s)
 (08/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_multiple_ns_modes_region: PASS (1.64 s)
 (09/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_multiple_ns_multiple_region: PASS (35.96 s)
 (10/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_nslot_namespace: PASS (5.68 s)
 (11/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_namespace_reconfigure: PASS (45.74 s)
 (12/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_check_namespace: PASS (2.07 s)
 (13/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_check_numa: PASS (1.74 s)
 (14/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_label_read_write: PASS (45.45 s)
 (15/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_daxctl_list: PASS (23.12 s)
 (16/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_sector_write: PASS (556.77 s)
 (17/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_fsdax_write: PASS (164.19 s) 
 (18/18) tests/avocado-misc-tests/memory/ndctl.py:NdctlTest.test_devdax_write: PASS (771.30 s)
RESULTS    : PASS 16 | ERROR 0 | FAIL 2 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
```

File: memory/ndctl.py [hunk 1]
```diff
@@ -251,20 +251,35 @@ def setUp(self):
...
         self.mode_to_use = self.params.get('modes', default='fsdax')
+        self.reflink = ''
 
-        if 'SuSE' not in self.dist.name:
+        if self.dist.name not in ['SuSE', 'rhel']:
             self.cancel('Unsupported OS %s' % self.dist.name)
+        else:
+            if self.dist.name == 'rhel':
+                # DAX wont work with reflink, disabling here
+                self.reflink = '-m reflink=0'
```
Comment (PraveenPenguin) [inline memory/ndctl.py]: on Suse we did not need this param ?

File: memory/ndctl.py [hunk 2]
```diff
@@ -279,15 +294,13 @@ def setUp(self):
             self.binary = './ndctl/ndctl'
             self.daxctl = './daxctl/daxctl'
         else:
-            deps.extend(['ndctl'])
+            for pkg in ['ndctl', 'daxctl']:
```
Comment (PraveenPenguin) [inline memory/ndctl.py]: :+1:

File: memory/ndctl.py [hunk 3]
```diff
@@ -431,20 +444,22 @@ def test_multiple_ns_multiple_region(self):
         self.disable_namespace()
         self.destroy_namespace()
         for val in regions:
+            ns_size = None
             region = self.get_json_val(val, 'dev')
             self.log.info("Using %s for muliple namespaces", region)
             self.log.info("Creating %s namespaces", self.cnt)
             if not self.size:
-                self.size = self.get_json_val(self.get_json(
```
Comment (PraveenPenguin) [inline memory/ndctl.py]: why variable name change requires is it for readability ?

File: memory/ndctl.py (hunk 1 above)
Comment (harish-24) [reply memory/ndctl.py]: Yes, we don't need.

File: memory/ndctl.py (hunk 3 above)
Comment (harish-24) [reply memory/ndctl.py]: It is being using inside the for loop as a check. As stated in commit message it has to be changed to a local variable to have a proper check throughout the loop.

**Pull request merged**